hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/replication/LegacyReplicationManager.java

```diff
@@ -1221,12 +1221,24 @@ private void handleUnderReplicatedContainer(final ContainerInfo container,
               id, misRepDelta);
         }
       } else {
+        replicas.stream().filter(rp -> rp.getState().equals(State.UNHEALTHY))
+            .forEach(rp -> {
+              sendDeleteCommand(container, rp.getDatanodeDetails(), true);
+              LOG.info("Send deleteCmd to force delete the unhealthy " +
+                  "replica {}", rp);
+            });
+        container.setNumberOfKeys(0);
+        container.setUsedBytes(0);
+        containerManager.updateContainerState(container.containerID(),
+            HddsProtos.LifeCycleEvent.DELETE);
         LOG.warn("Cannot replicate container {}, no healthy replica found.",
             container.containerID());
       }
     } catch (IOException | IllegalStateException ex) {
       LOG.warn("Exception while replicating container {}.",
           container.getContainerID(), ex);
+    } catch (InvalidStateTransitionException | TimeoutException e) {
+      e.printStackTrace();
```

Comment:
This "force delete" part looks a bit fishy, and was added in a commit with message `temp cleanup unhealthy container`.  Is this intended to be part of the final change?

Reply:
My mistake, this is my temp solution to the unhealthy containers, not the part of this patch.